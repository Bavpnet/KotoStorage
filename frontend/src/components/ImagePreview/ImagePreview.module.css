.img {
  width: 30vw;
  height: 30vw;
  object-fit: cover;
  border-radius: 10px;
}

.imgContainer {
  display: flex;
  flex-direction: column;
  gap: 38px;
  align-items: center;
  margin-top: 90px;
}

.heading3 {
  font-family: 'Graphik', sans-serif;
  font-size: 1rem;
  font-weight: 400;
}
