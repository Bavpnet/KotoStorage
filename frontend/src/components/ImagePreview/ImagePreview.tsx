import { SyntheticEvent, useEffect, useState } from 'react'
import styles from './ImagePreview.module.css'

type ImagePreviewProps = {
  url: string
}

export const ImagePreview: React.FC<ImagePreviewProps> = ({ url }) => {
  const url404 = 'https://http.cat/404'

  const onError = (e: SyntheticEvent<HTMLImageElement, Event>) => {
    console.log('error')
    if (e.currentTarget.src !== url404) {
      e.currentTarget.src = url404
    }
  }

  return (
    <picture className={styles.imgContainer}>
      <h3 className={styles.heading3}>Preview of the photo:</h3>
      <img className={styles.img} src={url} alt="preview" onError={onError} />
    </picture>
  )
}
