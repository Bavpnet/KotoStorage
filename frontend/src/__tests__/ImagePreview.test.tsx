import { render } from '@testing-library/react'
import ReactTestingUtils from 'react-dom/test-utils'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import { ImagePreview } from '../components/ImagePreview'

describe('ImagePreview', () => {
  it('renders correctly', () => {
    const setIsValidUrl = jest.fn()
    const url = 'https://http.cat/200'
    const tree = renderer
      .create(<ImagePreview url={url} setIsValidUrl={setIsValidUrl} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with invalid url', () => {
    const setIsValidUrl = jest.fn()
    const url = 'http://invalid.url'

    const { getByAltText } = render(
      <ImagePreview url={url} setIsValidUrl={setIsValidUrl} />
    )

    const image = getByAltText('preview') as HTMLImageElement

    ReactTestingUtils.Simulate.error(image)

    expect(image).toHaveAttribute('src', 'https://http.cat/404')
  })
})
