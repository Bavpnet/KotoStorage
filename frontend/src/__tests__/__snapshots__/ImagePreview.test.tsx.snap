// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview renders correctly 1`] = `
<picture
  className="imgContainer"
>
  <h3
    className="heading3"
  >
    Preview of the photo:
  </h3>
  <img
    alt="preview"
    className="img"
    onError={[Function]}
    onLoad={[Function]}
    src="https://http.cat/200"
  />
</picture>
`;
